src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp

```diff
@@ -872,14 +872,17 @@ namespace {
             for (const auto& well_name : well_names) {
                 this->updateWellStatus( well_name , handlerContext.currentStep , status, false, handlerContext.keyword.location() );
 
-                const auto table_nr = record.getItem("VFP_TABLE").get< int >(0);
                 std::optional<VFPProdTable::ALQ_TYPE> alq_type;
                 auto& dynamic_state = this->wells_static.at(well_name);
                 auto well2 = std::make_shared<Well>(*dynamic_state[handlerContext.currentStep]);
                 const bool switching_from_injector = !well2->isProducer();
                 auto properties = std::make_shared<Well::WellProductionProperties>(well2->getProductionProperties());
                 bool update_well = false;
 
+                auto table_nr = record.getItem("VFP_TABLE").get< int >(0);
```

Comment:
I think this should rather be:
```
if (record.getItem("VFP_TABLE").defaultApplied(0))
     table_nr = properties->VFPTableNumber;
```
Then we merge

Reply:
Done